artichoke-backend/src/extn/core/time/trampoline.rs

```diff
@@ -135,9 +94,11 @@ pub fn at(
         .checked_add(subsec_secs)
         .ok_or(ArgumentError::with_message("Time too large"))?;
 
-    let offset = match options {
-        Some(options) => offset_from_options(interp, options)?,
-        _ => Offset::local(),
+    let offset: Offset = if let Some(options) = options {
+        let offset: Option<Offset> = interp.try_convert_mut(options)?;
+        offset.unwrap_or(Offset::local())
```

Comment:
since `Offset::local` is potentially an expensive call, can we use `offset.unwrap_or_else(Offset::local)` instead?